### DISCUSSION TITLE
Cannot image any disks

### DISCUSSION THREAD

**Discussion by yuv422** [keirf/greaseweazle#379] [General]
I haven't had any success imaging any disks with my greaseweazle v4.

I have a 3.5" samsung floppy drive with a twisted cable and external power supply. I can seek the disk and see the drive spinning in read mode.

But I just get errors whenever I try to do a read.

```
./gw.exe read --format=ibm.scan test.img
Reading c=0-79:h=0-1 revs=2
Format ibm.scan
** FATAL ERROR:
Track0 signal absent after seek to cylinder 0

./gw.exe read --format=ibm.scan test.img
Reading c=0-79:h=0-1 revs=2
Format ibm.scan
Command Failed: Seek: Track 0 not found
```

The same drive, cable and disk works correctly on my 486 computer.  I can image disks just fine there. 

Any ideas what I could try to get it working with the greazeweazle?

**Comment by keirf** [comment]
It seems the drive isn't seeking reliably. You could try adjusting the step rate using the command `gw delays --step 3000` for example to increase the step rate to 3ms per step, or specify `20000` to slow it down to 20ms per step.

**Comment by keirf** [comment]
It might be that the pc is more forgiving of a sticky stepper motor or worm gear, for example.

**Comment by yuv422** [comment]
I gave it a try and it didn't make a difference.
I tried bumping all the delays a bit
```
./gw.exe delays
Select Delay: 20us
Step Delay: 10000us
Settle Time: 60ms
Motor Delay: 2000ms
Watchdog: 10000ms
```

I did notice that if I do a `seek 0` it moves the head to the edge of the disk. If I then do a read it will move the head slightly towards the center of the disk. It then gives me the error. If I then try to do another `seek 0` I get an error. 
If I seek to a different location it works and I can then seek back to 0.

```
Eric@DESKTOP-MINGW64 ~/greaseweazle-1.14
$ ./gw.exe seek 3

Eric@DESKTOP-MINGW64 ~/greaseweazle-1.14
$ ./gw.exe seek 0

Eric@DESKTOP-MINGW64 ~/greaseweazle-1.14
$ ./gw.exe seek 0

Eric@DESKTOP-MINGW64 ~/greaseweazle-1.14
$ ./gw.exe read --format=ibm.scan test.img
Reading c=0-79:h=0-1 revs=2
Format ibm.scan
** FATAL ERROR:
Track0 signal absent after seek to cylinder 0

Eric@DESKTOP-MINGW64 ~/greaseweazle-1.14
$ ./gw.exe seek 0
** FATAL ERROR:
Track0 signal absent after seek to cylinder 0

Eric@DESKTOP-MINGW64 ~/greaseweazle-1.14
$ ./gw.exe seek 1

Eric@DESKTOP-MINGW64 ~/greaseweazle-1.14
$ ./gw.exe seek 0
```

**Reply by keirf** [reply]
Maybe the track 0 sensor is somehow unreliable or marginal. It's just a mechanically secured photo sensor.

**Reply by yuv422** [reply]
Good point. I'll see if I can get another drive to test.

**Reply by aurum79argentum47** [reply]
My Shugart SA400L does exactly the same thing.   Brand new (new old stock) photodiode and photsensor for track 0 sensor, calibrated per service manual.

Switches do not help.

**Reply by keirf** [reply]
Have you tried increasing the step delay to 20000us or beyond? Manual says the step latency is 20ms on this drive.

**Reply by aurum79argentum47** [reply]
Yes, I've tried this after reading the previous posts.  I'm about to give up on these two old Shugart 400's with GW.  They work great on the TI-99/4A but maybe they are just to sloppy for this precise controller.

**Comment by Zalorpas** [comment]
Could you, please, share the Samsung FDD model? It might help for the troubleshooting
Thanks,

**Reply by aurum79argentum47** [reply]
I'm not the Samsung guy.   I tried a Mitsubishi M504C-368U (5-1/4" HD) with sucess.
